src/Log/Engine/BaseLog.php

```diff
@@ -57,13 +57,16 @@ public function __construct(array $config = [])
     {
         $this->setConfig($config);
 
-        if (!is_array($this->_config['scopes']) && $this->_config['scopes'] !== false) {
+        assert(
```

Comment:
Are you thinking that we should use `assert()` more for developer error checking? Would we also start recommending that people use `assert_options()` to optimize away assertion checks in production?

Reply:
I'm not sure we should use assertions where cake can't continue. If the data is invalid, then we should probably just throw the exception which is guaranteed.

Reply:
> Are you thinking that we should use assert() more for developer error checking? 

Yup, so that the checks/safeguards have zero impact in production. 

> Would we also start recommending that people use assert_options() to optimize away assertion checks in production?

The manual actually recommends using the ini config `zend.assertions` instead of `assert_options()` in PHP 7+ code.

> I'm not sure we should use assertions where cake can't continue. If the data is invalid, then we should probably just throw the exception which is guaranteed.

IMO `assert()` is the right thing to use in this case. The check is only there to avoid missing updating the config when upgrading from 4.x.

Reply:
Sounds good. We could probably check !is_bool() to be comprehensive.